app/services/structured_fact_importer.rb

```diff
@@ -24,19 +24,21 @@ def preload_fact_names
     Hash[fact_name_class.where(:type => fact_name_class).reorder('').pluck(:name, :id, :compose).map { |fact| [fact.shift, fact] }]
   end
 
-  def create_fact_name(fact_names, name, fact_value)
+  def find_or_create_fact_name(name, value)
```

Comment:
Make `def .... (name, value=nil)` , it's more intuitive  as it just creates the FactName w/o passing nil as you do in line https://github.com/theforeman/foreman/pull/3667/files#diff-509173a705422d0025bd1a6c8a630c72R30